raiden_contracts/tests/test_contract_joiner.py

```diff
@@ -0,0 +1,28 @@
+from tempfile import NamedTemporaryFile
+
+import pytest
+
+from raiden_contracts.utils.join_contracts import ContractJoiner
+
+
+def test_contract_joiner_with_non_existent_import() -> None:
+    """ Using ContractJoiner on a source importing a nonexistent file """
+    joiner = ContractJoiner()
+    source_with_missing_import = """
+        import "NonExistent.sol";
+    """
+    with NamedTemporaryFile() as source_file:
+        source_file.write(bytearray(source_with_missing_import, "ascii"))
+        source_file.flush()
+        with pytest.raises(FileNotFoundError):
+            joiner.join(open(source_file.name))
+
+
+def test_contract_joiner_with_empty_file() -> None:
+    """ Using ContractJoiner on an empty source """
+    joiner = ContractJoiner()
+    empty_source = ""
+    with NamedTemporaryFile() as source_file:
+        source_file.write(bytearray(empty_source, "ascii"))
+        source_file.flush()
```

Comment:
You don't need to explicitly write nothing to the source file. It is already empty before.

Reply:
Copy-paste-modify did harm here.  I'll remove these lines and the empty string.